src/Console/AddCommand.php

```diff
@@ -138,24 +138,33 @@ private function getProperty(
                 new Question("$desc: ", $default)
             );
         }
-        //$value = $value ?: "";
         return $value;
     }
 
     private function setPayerNumber(
         string $value,
         DonorBuilder $donorBuilder
     ) {
-        $donorBuilder->setPayerNumber($value);
+        if (is_numeric($value) && strlen($value) <= 16) {
+            $donorBuilder->setPayerNumber($value);
+        } else {
+            throw new \Exception('Payer number must be numerical, and max 16 digits');
+        }
     }
 
     private function setAccount(
         $value,
         DonorBuilder $donorBuilder,
         AccountFactory $accountFactory
     ) {
-        $newAccount = $accountFactory->createAccount($value);
-        $donorBuilder->setAccount($newAccount);
+        //TODO: the accountFactory seems quite liberal in what values it
```

Comment:
Actually no accident. The factory removes non-didigts in a pre-processing pass. And since your test number does not contain a valid clearing number it will be parsed as an `UnknownAccount`, and thus we have no checkdigit or structure to validate.

If you pass a number with a clearing number, like `33007501121234` you wil get an error (if I not accidently just wrote the correct check digit...).

All of this can be configured when we create the factory, the pre-processing step can be removed, and we can blacklist the unknown account if we want.